src/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# libcps
+add_library(libcps cps/env.cpp cps/loader.cpp cps/printer.cpp cps/search.cpp
+                   cps/utils.cpp cps/version.cpp)
+
+set(CMAKE_CXX_STANDARD 23)
+
+# Configure config.hpp
+set(functions unreachable)
+set(headers utility)
+set(BUILTIN_FEATURE_MACROS "")
+include(CheckCXXSymbolExists)
+foreach(function header IN ZIP_LISTS functions headers)
+  string(TOUPPER ${function} function_upper)
+  check_cxx_symbol_exists(std::${function} ${header}
+                          HAS_BUILTIN_${function_upper})
+  if(${HAS_BUILTIN_${function_upper}})
+    set(BUILTIN_FEATURE_MACROS
+        "${BUILTIN_FEATURE_MACROS}\n#define CPS_USE_BUILTIN_${function_upper} 1"
+    )
+  else()
+    set(BUILTIN_FEATURE_MACROS
+        "${BUILTIN_FEATURE_MACROS}\n#define CPS_USE_BUILTIN_${function_upper} 0"
+    )
+  endif()
```

Comment:
This feels like a lot of repetition, CMake doesn't have a builtin mechanism to turn a bool into an int, or even a ternary?

Reply:
The only one I know of is https://cmake.org/cmake/help/latest/command/configure_file.html#transformations. I'm not sure how to make it work with an dynamic number of macros.

Reply:
Google suggests that maybe something like `$<IF:${HAS_BUILTIN_${function_upper}},1,0>`, but maybe some of the more CMake fluent people have suggestions.

Reply:
Oh sorry I misunderstood your original comment. I was thinking of something else. Getting rid of the if else block